apps/andi/lib/andi/dataset_cache.ex

```diff
@@ -4,26 +4,46 @@ defmodule Andi.DatasetCache do
 
   require Logger
 
-  alias Andi.Services.DatasetRetrieval
+  import Andi, only: [instance_name: 0]
 
-  # Client
   def start_link(_) do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
-  def put_datasets(datasets) when is_list(datasets) do
-    Enum.each(datasets, &put_dataset/1)
+  def put(datasets) when is_list(datasets) do
+    Enum.each(datasets, &put/1)
   end
 
-  def put_dataset(%SmartCity.Dataset{} = dataset) do
-    :ets.insert(__MODULE__, {dataset.id, dataset})
+  def put(%SmartCity.Dataset{} = dataset) do
+    updated =
+      dataset.id
+      |> get()
+      |> Map.merge(%{"id" => dataset.id, "dataset" => dataset})
+
+    :ets.insert(__MODULE__, {dataset.id, updated})
+  end
+
+  def put(%{"id" => id, "ingested_time" => time_stamp}) do
```

Comment:
This feels like it should be a separate function rather than just a function head on put.

Reply:
These are all putting to the cache with different structures (but the same result) and so different function heads with pattern matching makes the most sense to me.

Reply:
I think it would be better to have an `update_ingested_time` function.

You're also not "putting", you're fetching and existing one and modifying a field on it.